File: hiresignal-proto/backend/app/schemas/resume.py
from pydantic import BaseModel, Field
from typing import Optional, List
from uuid import UUID


class ResumeResultResponse(BaseModel):
    id: UUID
    filename: str
    final_score: float
    score_semantic: float
    score_tfidf: float
    score_skills: float
    score_experience: float
    rank: Optional[int]
    matched_skills: List[str]
    missing_skills: List[str]
    years_experience_detected: Optional[float]
    language_detected: str
    extraction_quality: str
    flags: dict
    created_at: str

    class Config:
        from_attributes = True


class SingleScreenRequest(BaseModel):
    jd_text: str


class SingleScreenResponse(BaseModel):
    filename: str = "resume"
    final_score: float
    score_semantic: float
    score_tfidf: float
    score_skills: float
    score_experience: float
    matched_skills: List[str] = Field(default_factory=list)
    missing_skills: List[str] = Field(default_factory=list)
    years_experience_detected: Optional[float] = None
    language_detected: str = "en"
    extraction_quality: str = "good"
    flags: dict = Field(default_factory=dict)
    processing_ms: Optional[int] = None

File: hiresignal-proto/frontend/src/types/resume.ts
export interface ResumeResult {
  id: string
  job_id: string
  tenant_id: string
  filename: string
  file_key: string
  content_hash: string
  score_semantic: number
  score_tfidf: number
  score_skills: number
  score_experience: number
  final_score: number
  extracted_text: string
  extraction_quality: 'good' | 'ocr_used' | 'poor'
  language_detected: string
  extracted_text_length: number
  matched_skills: string[]
  missing_skills: string[]
  years_experience_detected?: number
  flags: Record<string, boolean>
  status: 'pending' | 'done' | 'failed'
  error_message?: string
  processing_ms?: number
  rank?: number
  created_at: string
  updated_at: string
}

export interface ResumeResultResponse extends ResumeResult {}

export interface SingleScreenRequest {
  jd_text: string
  resume_file?: File
}

export interface SingleScreenResponse {
  final_score: number
  score_semantic: number
  score_tfidf: number
  score_skills: number
  score_experience: number
  matched_skills: string[]
  missing_skills: string[]
  years_experience_detected?: number
  flags: Record<string, boolean>
}

export interface FeedbackCreate {
  result_id: string
  action: 'shortlist' | 'reject'
  notes?: string
}

export interface Feedback {
  id: string
  result_id: string
  job_id: string
  tenant_id: string
  user_id: string
  action: 'shortlist' | 'reject'
  notes?: string
  created_at: string
}
